tools/test/CMakeLists.txt

```diff
@@ -50,3 +50,35 @@ iree_lit_test_suite(
 )
 
 ### BAZEL_TO_CMAKE_PRESERVES_ALL_CONTENT_BELOW_THIS_LINE ###
+
+if((IREE_TARGET_BACKEND_LLVM_CPU OR DEFINED IREE_HOST_BINARY_ROOT) AND
+   IREE_HAL_DRIVER_LOCAL_SYNC)
+  iree_bytecode_module(
+    NAME
+      iree_run_module_bytecode_module_llvm_cpu
+    SRC
+      "iree-run-module.mlir"
+    MODULE_FILE_NAME
+      "iree_run_module_bytecode_module_llvm_cpu.vmfb"
+    C_IDENTIFIER
+      "iree_tools_test_iree_run_module_bytecode_module_llvm_cpu"
+    FLAGS
+      "--iree-hal-target-backends=llvm-cpu"
+  )
+
+  iree_run_module_test(
+    NAME
+      iree_run_module_correctness_test
+    MODULE_SRC
+      "${CMAKE_CURRENT_BINARY_DIR}/iree_run_module_bytecode_module_llvm_cpu.vmfb"
+    DRIVER
+      "local-sync"
+    RUNNER_ARGS
+      "--entry_function=abs"
+      "--function_input=f32=-10"
+    EXPECTED_OUTPUT
+      "f32=10"
+    DEPS
+      ::iree_run_module_bytecode_module_llvm_cpu_c
```

Comment:
Can this depend on the .vmfb target instead? (Seems hacky to generate a C wrapper just for the sake of having a target to add to `DEPS`, if that's what this is doing)

Reply:
`iree_bytecode_module` is kind of hacky today. `.vmfb` is not built as a custom target, but a command, which is actually attached to the `iree_cc_library` that is called from `iree_c_embed_data`

https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_bytecode_module.cmake#L125
https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_c_embed_data.cmake#L86

The `C_IDENTIFIER` is needed so `iree_c_embed_data` can be called.
https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_bytecode_module.cmake#L156-L174

We can try to clean this up in a different PR. :(

Reply:
@pzread fixed this. There's now a custom target for the vmfb itself: https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_bytecode_module.cmake#L145

Reply:
Ahh.. Sorry I missed that. Cleaned up `iree_run_module_correctness_test` and its dependency.